cities_light/management/commands/cities_light.py

```diff
@@ -360,6 +380,79 @@ def region_import(self, items):
                 force_update=force_update
             )
 
+    def subregion_import(self, items):
+        try:
+            subregion_items_pre_import.send(sender=self, items=items)
+        except InvalidItems:
+            return
+
+        try:
+            force_insert = False
+            force_update = False
+            subregion = SubRegion.objects.get(geoname_id=items[ISubRegion.geonameid])
+            force_update = True
+        except SubRegion.DoesNotExist:
+            if self.noinsert:
+                return
+            subregion = SubRegion(geoname_id=items[ISubRegion.geonameid])
+            force_insert = True
```

Comment:
Have you considered something like this ?

```
force_insert = force_update = False
subregion = SubRegion.objects.filter(geoname_id=items[ISubRegion.geonameid]).first()
if subregion:
    force_update = True
elif not subregion and self.noinsert:
    return
else:
    subregion = SubRegion(geoname_id=items[ISubRegion.geonameid])
    force_insert = True
```

Reply:
Not really. I let you improve or fix things as you think it's better.